docs/flow/config-language/components.md

```diff
@@ -6,3 +6,96 @@ weight: 300
 ---
 
 # Components
+Components are the defining feature of Grafana Agent Flow. They are autonomous
+pieces of business logic that perform a single task (like retrieving secrets or
+collecting Prometheus metrics) and can be wired together to form programmable
+pipelines of telemetry data.
+
+Under the hood, components are orchestrated via the [_component
+controller_]({{< relref "../concepts/component_controller.md" >}}), who is
+responsible for scheduling them, reporting their health and debug status, and
+continuously re-evaluating their inputs and outputs.
```

Comment:
I'd like to avoid the word "continuous" here, since it sounds like components are always being reevaluated no matter what, implying that it might be heavy on CPU usage.

Can we change this to say that the component controller determines when components need to be reevaluated?

Reply:
(also, nit: exports don't get evaluated, only arguments do)